README.md

```diff
@@ -1,3 +1,65 @@
 # css-modules-typescript-loader
 
-Webpack loader to create type declarations for css modules
+[Webpack](https://webpack.js.org/) loader to create [TypeScript](https://www.typescriptlang.org/) declarations for [CSS Modules](https://github.com/css-modules/css-modules).
+
+## Why?
+
+There are currently a lot of [solutions to this problem](https://www.npmjs.com/search?q=css%20modules%20typescript%20loader). However, this package differs in the following ways:
+
+- Encourages generated TypeScript declarations to be checked into source control, which allows `webpack` and `tsc` commands to be run in parallel in CI.
+
+- Ensures committed TypeScript declarations are in sync with the code that generated them via the [`verify` mode](#verify-mode).
+
+- Doesn't silently ignore invalid TypeScript identifiers. If a class name is not valid TypeScript (e.g. `.foo-bar`, instead of `.fooBar`), `tsc` should report an error.
+
+## Usage
+
+This loader emits TypeScript declaration files matching your CSS Modules e.g. `Component.css.d.ts`. Place `css-modules-typescript-loader` directly after `css-loader` in your webpack config.
```

Comment:
The first sentence doesn't really belong here, IMO, since it's not "usage".
```suggestion
Place `css-modules-typescript-loader` directly after `css-loader` in your webpack config.
```